api/lanes.py

```diff
@@ -1051,13 +1057,58 @@ def order_facet_to_database_field(cls, order_facet):
         else:
             return Facets.order_facet_to_database_field(order_facet)
 
+
+class CrawlableCollectionBasedLane(DynamicLane):
+
+    def __init__(self, library, collections=None):
+        """Create a lane that finds all books in the given collections.
+
+        :param library: The Library to use for purposes of annotating
+            this Lane's OPDS feed.
+        :param collections: A list of Collections. If none are specified,
+            all Collections associated with `library` will be used.
+        """
+        self.library_id = library.id
+        if collections:
+            identifier = " / ".join(sorted([x.name for x in collections]))
+        else:
+            identifier = library.name
+            collections = library.collections
+        self.initialize(library, "Crawlable feed: %s" % identifier)
+        if collections:
+            # initialize() set the collection IDs to all collections
+            # associated with the library. We may want to restrict that
+            # further.
+            self.collection_ids = [x.id for x in collections]
+
+    def get_library(self, _db):
+        """Find the Library object associated with this WorkList."""
+        return Library.by_id(_db, self.library_id)
+
+    def bibliographic_filter_clause(self, _db, qu, featured=False):
+        """Filter out any books that aren't in the right collections."""
+        # The normal behavior of works() is to put a restriction on
+        # collection_ids, so we only need to do something if
+        # there are no collections specified.
+        if not self.collection_ids:
+            # When no collection IDs are specified, there is no lane
+            # whatsoever
+            return None, None
+        return super(
+            CrawlableCollectionBasedLane, self).bibliographic_filter_clause(
+                _db, qu, featured
+            )
+
+
 class CrawlableCustomListBasedLane(DynamicLane):
     """A lane that consists of all works in a single CustomList."""
 
     uses_customlists = True
 
     def initialize(self, library, list):
-        super(CrawlableCustomListBasedLane, self).initialize(library, list.name)
+        super(CrawlableCustomListBasedLane, self).initialize(
+            library, "Crawlable feed: %s" % list.name
+        )
```

Comment:
Making sure every crawlable feed has "Crawlable" in its identifier guarantees that it won't clash with other types of feeds when it's cached using the `CacheFeed` code.